src/main/java/net/revelc/code/formatter/FormatterMojo.java

```diff
@@ -478,9 +493,18 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             final var elapsed = TimeUtil.printDuration(duration);
 
             final String results = String.format(
-                    "Processed %d files in %s (Formatted: %d, Unchanged: %d, Failed: %d, Readonly: %d)", numberOfFiles,
-                    elapsed, rc.successCount, rc.skippedCount, rc.failCount, rc.readOnlyCount);
-            this.getLog().info(results);
+                    "Processed %d files in %s (Formatted: %d, Skipped: %d, Unchanged: %d, Failed: %d, Readonly: %d)",
+                    numberOfFiles, elapsed, rc.successCount, rc.skippedCount, rc.unchangedCount, rc.failCount,
+                    rc.readOnlyCount);
+
+            if (rc.failCount > 0)
+                this.getLog().error(results);
+            else if (rc.readOnlyCount > 0)
+                this.getLog().warn(results);
+            else if (warnUnchanged && rc.successCount == 0 && rc.skippedCount == 0)
+                this.getLog().warn(results);
+            else
+                this.getLog().info(results);
```

Comment:
I don't think these should merely log the regular results to the warning color. The results should still show on INFO. But, if you want to add a log *message* for the specific cases, in addition to the regular results at the INFO level, then I think that would make more sense (with the exception of the warning when there's no work to do, as I don't think that's useful).

Reply:
I don't agree with that. The summary line should be logged at the level of the most severe result. Just look at Maven itself. I've added a log per warning/error.

Reply:
Mainly my concern would be that sometimes WARN/ERROR are configured by the logging systems to be redirected to STDERR, and it would be harder to grep STDOUT for the results, or to do other automated processing if you don't know where they are going. However, I don't feel too strongly about it, and I like the other logging changes, so I'm fine with this.